File: Chapter09/Java/LicenseCheck/app/src/main/java/net/zenconsult/android/CommercialAppActivity.java
package net.zenconsult.android;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.widget.Button;
import android.widget.TextView;

public class CommercialAppActivity extends AppCompatActivity implements CommsEvent {

    private static final String TAG = CommercialAppActivity.class.getSimpleName();

    private TextView view;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        view = findViewById(R.id.editText1);

        // Verify license on first run

        // Click Button
        final Button button = findViewById(R.id.button1);
        button.setOnClickListener((v) -> {
            view.setText("Fetching fact...");
            CommsNotifier c = new CommsNotifier(this);
            c.start();
        });
    }

    @Override
    public void onTextReceived(String text) {
        runOnUiThread(() -> {
            view.setText(text);
        });
    }
}

File: Chapter09/Kotlin/LicenseCheck/app/src/main/java/net/zenconsult/android/CommercialAppActivity.kt
package net.zenconsult.android

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.TextView

class CommercialAppActivity : AppCompatActivity(), CommsEvent {

    private lateinit var view: TextView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        view = findViewById(R.id.editText1)

        // Verify license on first run

        // Click Button
        val button = findViewById<Button>(R.id.button1)
        button.setOnClickListener {
            view.text = "Fetching fact..."
            val c = CommsNotifier(this)
            c.start()
        }
    }

    override fun onTextReceived(text: String) {
        TODO("Not yet implemented")
    }

    companion object {
        const val TAG = "CommercialAppActivity"
    }
}